### ISSUE TITLE
False positive for WPS323 on date/time formatting

### ISSUE THREAD

**Issue by vmozhaev** [wemake-services/wemake-python-styleguide#1329]
# Bug report

Some date and time formatting cases are treated as violation:

    some_datetime.strftime('%Y-%m-%d-%H-%M')
    '{0:%Y-%m-%d-%H-%M}'.format(some_datetime)

But others are not:

    some_datetime.strftime('%H-%M')
    '{0:%H-%M}'.format(some_datetime)

## What's wrong

There is no easy to use alternative for custom date/time formatting except C standard codes.

## How is that should be

The "%" symbol should be allowed at least as part of format specification for new-style format strings.

**Comment by sobolevn** [comment]
Workaround can be: if there's also any datespecific `%` modifier as `%Y` or `%m` we can ignore this case.

**Comment by Jrryy** [comment]
If I'm allowed to, I'd like to try and find some way to solve this issue.

**Comment by sobolevn** [comment]
You are more than welcome! In case you have any questions, please ask me 🙂

**Comment by agatti** [comment]
How difficult would it be to extend this to also ignore SQL statements?  For example, with [psycopg2](https://www.psycopg.org) passing parameters to SQL queries is done by using `%s` markers in the query (as in `SELECT * FROM table WHERE column = %s`, as per [the documentation](https://www.psycopg.org/docs/usage.html#passing-parameters-to-sql-queries)).  

With code doing a lot of SQL stuff it may become sort of tedious to add all the `noqa` markers on each and every query (multiline or not)...

**Comment by Jrryy** [comment]
I have not used raw psycopg2 so I am not sure about what I am about to say: if the only function that uses `%s` is `cur.execute()`, I can ignore those `%s` as long as they are being used inside of that function. In other cases, let's say that you save the query in a variable and then use it in `cur.execute()`, it will be pretty difficult for me to come up with a way to ignore that because I would need to check the entire AST of a project to finally see if that variable is being used in that function or it is simply being formatted with the `%` operator.

To sum it up: adding simple exceptions such as that `%s` is being used inside `cur.execute()` or that `%Y`, `%m`... are for datetime formatting is possible, but not 100% reliable, as you might expect.

**Comment by agatti** [comment]
Fair enough, I was just wondering on the feasibility :)

**Comment by Jrryy** [comment]
Well it is feasible, I just can't tell if the solution I proposed will be accepted because it's just a workaround after all, but as for now I can't think of a better way to do it.
